src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -312,12 +408,235 @@ public static String getSampleNameFromReadCounts(final ReadCountCollection count
+        final Map<String, Double> columnsToDrop = new LinkedHashMap<>((int) (extremeColumnMedianCountPercentileThreshold * 4.0 / 100.0));
+        for (int i = 0; i < columnMedians.length; i++) {
+            if (columnMedians[i] >= bottomExtremeThreshold && columnMedians[i] <= topExtremeThreshold) {
+                columnsToKeep.add(columnNames.get(i));
+            } else {
+                columnsToDrop.put(columnNames.get(i), columnMedians[i]);
+            }
+        }
 
+        // log and drop columns if it applies
+        if (columnsToKeep.isEmpty()) {
+            throw new UserException.BadInput("No column count left after applying the extreme counts outlier filter");
+        } else if (columnsToKeep.size() == columnNames.size()) {
+            logger.info(String.format("No column dropped due to extreme counts outside [%.10f, %.10f]", bottomExtremeThreshold, topExtremeThreshold));
+            return readCounts;
+        } else {
+            logger.info(String.format("Some columns dropped, %d of %d, as they are classified as having extreme median counts across targets outside [%.10f, %.10f]: e.g. %s",
+                    columnNames.size() - columnsToKeep.size(), columnNames.size(), bottomExtremeThreshold, topExtremeThreshold,
+                    columnsToDrop.entrySet().stream().limit(10).map(kv -> kv.getKey() + " (" + kv.getValue() + ")").collect(Collectors.joining(", "))));
+            return readCounts.subsetColumns(columnsToKeep);
+        }
+
+    }
+
+    private static long countZeroes(final double[] data) {
+        return DoubleStream.of(data).filter(d -> d == 0.0).count();
+    }
+
+    /**
+     * Remove targets that have too many counts equal to 0.
+     * <p>
+     *     It will return a copy of the input read-count collection with such targets dropped.
+     * </p>
+     *
+     * @param readCounts the input read counts.
+     * @param maximumTargetZeros maximum number of counts equal to 0. per target tolerated.
+     * @return never {@code null}. It might be a reference to the input read-counts if there is
+     *   is no target to be dropped.
+     */
+    public static ReadCountCollection removeTargetsWithTooManyZeros(final ReadCountCollection readCounts, final int maximumTargetZeros, final Logger logger) {
+
+        final RealMatrix counts = readCounts.counts();
+
+        final Set<Target> targetsToKeep = IntStream.range(0, counts.getRowDimension()).boxed()
+                .filter(i -> countZeroes(counts.getRow(i)) <= maximumTargetZeros)
+                .map(i -> readCounts.targets().get(i)).collect(Collectors.toSet());
+
+        final int targetsToDropCount = readCounts.targets().size() - targetsToKeep.size();
+        if (targetsToDropCount == 0) {
+            logger.info(
+                    String.format("No target drop due to too many zero counts; there is no target with a large number of columns with zero counts (<= %d of %d) ", maximumTargetZeros, readCounts.columnNames().size()));
```

Comment:
No **targets were dropped** due to too many zero counts;  
Also second part of this sentence sounds redundant